materials/06_statistical_analysis.Rmd

```diff
@@ -331,379 +327,308 @@ limma_results %>%
+post_hoc_sig <- my_results %>%
+  filter(significance == 'sig') %>%
+  pull(Protein)
+
+treat_not_sig <- limma_results_treat %>% 
+  filter(significance == 'not.sig') %>%
+  pull(Protein)
+
+sig_post_hoc_only <- intersect(post_hoc_sig, treat_not_sig)
+
+# Make a volcano plot highlighting these proteins
+p <- my_results %>%
+  # Add a new column to annotate the proteins to highlight
+  mutate(highlight = Protein %in% sig_post_hoc_only) %>% 
+  # Use the highlight column to control the shape aesthetic
+  ggplot(aes(x = logFC, y = -log10(P.Value),
+             fill = significance,
+             shape = highlight)) + 
+  geom_point(stroke = 0.25, size = 3) +
+  # Define the shapes
+  scale_shape_manual(values = c(21,8),
+                     name='Post-hoc logFC\nthresh. sig. only') + 
+  theme_bw()
 
-```
+print(p)
 
-Now let's plot the results
+# Re-use the plot object with updated data using the '%+%' operator.
+p2 <- p %+%
+  mutate(highlight=Protein %in% sig_post_hoc_only, limma_results_treat)
+print(p2)
 
-```{r}
-my_results %>%
-  ggplot(aes(x = logFC, y = -log10(P.Value), fill = result)) +
-  geom_point(shape = 21, stroke = 0.25, size = 3) +
-  theme_bw()
 ```
+:::
 
 
-:::
 
+## Visualising the protein abundances in a heatmap
 
-Another widely used visualisation tool are heatmaps. A heatmap is a two-dimensional
-representation of our quantitative data where the magniture of values are depicted
+Another widely used visualisation tool is a heatmap. A heatmap is a two-dimensional
+representation of our quantitative data where the magnitude of values are depicted
 by colour. These visualisations are commonly combined with clustering tools to
 facilitate the identification of groups of features, here proteins, that display
 similar quantitative behaviour. Here, we will use the `pheatmap` function from the 
 `pheatmap` package to plot a heatmap of proteins that display a significant 
-difference in abundance between M-phase and desynchronised cells.
+difference in abundance between M-phase and G1-phase cells. Note that we will plot all samples though, not just M and G1 phase samples.
 
 We first extract the accessions of proteins with significant differences. We 
-use these accessions to subset the original quantitation data which is currently
+use these accessions to subset the original quantification data which is currently
 stored in the `assay` of our `cp_qf` object.
 
 ```{r}
 ## Extract accessions of significant proteins
-sig_proteins <- M_Desynch_results %>%
+sig_proteins <- limma_results %>%
   filter(significance == "sig") %>%
-  pull(protein)
-
-## Save indices of rows where these proteins our in our cp_qf object
-sig_indices <- which(rowData(cc_qf[["log_norm_proteins"]])$Master.Protein.Accessions %in% sig_proteins)
+  pull(Protein)
 
 ## Subset quantitative data corresponding to significant proteins
-quant_data <- cc_qf[["log_norm_proteins"]][sig_indices, ] %>%
+quant_data <- cc_qf[["log_norm_proteins"]][sig_proteins, ] %>%
```

Comment:
I think it makes more sense to plot just the sig proteins. I've split over two lines to avoid the dreaded subset chaining:

```r
## Subset quantitative data corresponding to significant proteins
quant_data <- cc_qf[["log_norm_proteins"]]

quant_data <- quant_data[sig_proteins, ] %>% assay() 
```